lesson7/task3/src/Main.java

```diff
@@ -0,0 +1,97 @@
+import java.util.Arrays;
+import java.util.OptionalInt;
+
+public class Main {
+    int count = 0;
+    int numSentence;
+    int[] countText;
+
+
+
+   public void PalindromWord(StringBuilder word) {
+       String a = word.toString();
+       StringBuilder str = word.reverse();
+
+       if ((str.toString()).equalsIgnoreCase(a)) {
+           System.out.println("Слово " + a + " является палиндромом");
+                  } else {
+           System.out.println(a+" не палиндром");}
+   }
+
+    public int PalindromCount(StringBuilder sentence) { //определение количества полиндромов в предложении
+
+        String[] words = sentence.toString().split("\\s"); //массив из слов
+        StringBuilder str = sentence.reverse();//переворачиваем всё предложение
+        String[] wordsRevers = str.toString().split("\\s");//массив перевернутых слов
+        int count = 0;
+        for (int i = 0; i < words.length; i++) {
+            for (int j = words.length-1; j > -1; j--)// обратный порядок, т.к. перевернули предложение задом наперед
```

Comment:
Необходимо бежать одновременно по двум массивам, а не во вложенном цикле.
```suggestion
            for (int j = words.length-1, int i = 0; j > -1; j--, i++)
```